restricted_site_access.php

```diff
@@ -245,10 +245,16 @@ public static function restrict_access( $wp ) {
 
 		switch ( $rsa_restrict_approach ) {
 			case 4:
-				if ( ! empty( self::$rsa_options['page'] ) && ( $page_id = get_post_field( 'ID', self::$rsa_options['page'] ) ) ) {
-					unset( $wp->query_vars );
-					$wp->query_vars['page_id'] = $page_id;
-					return;
+				if ( ! empty( self::$rsa_options['page'] ) ) {
+					$page = get_post( self::$rsa_options['page'] );
```

Comment:
Just wondering if we need a check on $page before using it?

Reply:
Added with a fallback to default values in https://github.com/10up/restricted-site-access/pull/30/commits/4a15454a725426fd11bd176264c710d886f078a1